### PR TITLE
Make GitPod works

### PR DESCRIPTION
**Pull request by noklam**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

The whole Gitpod settings is not working at all

* installation is two steps, which makes the installed library very different from what you will get in the CI
* make build-docs cannot be run - error
* `pre-commit` is broken, it stuck with some corrupted virtualenv and you need to reinstall it manually.

Related to #https://github.com/kedro-org/kedro/issues/2586, but not a complete fix.
## Development notes

- `make lint` work nows again
- `make build-docs` works 
## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by noklam** [comment]
Pinging @heber-urdaneta @mobilize-mrojas @orellabac @Vladimir-Filimonov as I found you are the original author of this PR https://github.com/kedro-org/kedro-plugins/pull/104

**Comment by noklam** [comment]
@astrojuanlu I removed some of the setup and make it simpler. The current setup simply doesn't work any more as we are not running in the pre-build mode.

Do you have any idea to make the current setup to a Docker setup? This PR would at least make sure `pre-commit` works and the libraries installed properly. `make build-docs` is still not working unfortunately with `VideoDataSet` having some more complicated cv dependency, I haven't got time to fix this yet.

**Comment by noklam** [comment]
```
ImportError: libGL.so.1: cannot open shared object file: No such file or directory
>>> from kedro_datasets.video import *
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/workspace/.pyenv_mirror/user/current/lib/python3.8/site-packages/kedro_datasets/video/__init__.py", line 5, in <module>
    from .video_dataset import VideoDataSet
  File "/workspace/.pyenv_mirror/user/current/lib/python3.8/site-packages/kedro_datasets/video/video_dataset.py", line 12, in <module>
    import cv2
  File "/workspace/.pyenv_mirror/user/current/lib/python3.8/site-packages/cv2/__init__.py", line 8, in <module>
    from .cv2 import *
ImportError: libGL.so.1: cannot open shared object file: No such file or directory
```

**Comment by astrojuanlu** [comment]
👀 Let me have a look

**Comment by astrojuanlu** [comment]
- Right after the initial task finishes, `kedro_datasets` is not installed.
- `make build-docs` installs `kedro-datasets[all]~=1.4.0` (pulled from the `docs` optional dependencies group) and immediately fails on `VideoDataSet`
- Then `sudo apt-get update && sudo apt-get install -y --no-install-recommends libgl1` as advised in https://stackoverflow.com/a/74501248/554319 fixes the import, and `make build-docs` gets back to _almost_ working, but fails when generating the output because `mmdc` (the Mermaid CLI) is not installed
  - So, this setup needs either NPM preinstalled, or make use of conda environments. Maybe worth going back to the `full` image?

**Comment by noklam** [comment]
> * Right after the initial task finishes, `kedro_datasets` is not installed.
Do we need `kedro_datasets` installed for `kedro`?

> * `make build-docs` installs `kedro-datasets[all]~=1.4.0` (pulled from the `docs` optional dependencies group) and immediately fails on `VideoDataSet`
> * Then `sudo apt-get update && sudo apt-get install -y --no-install-recommends libgl1` as advised in https://stackoverflow.com/a/74501248/554319 fixes the import, and `make build-docs` gets back to _almost_ working, but fails when generating the output because `mmdc` (the Mermaid CLI) is not installed

`sudo` requires the Dockerfile? I try running sudo in terminal it doesn't work.

>   * So, this setup needs either NPM preinstalled, or make use of conda environments. Maybe worth going back to the `full` image?

I think that's fine.

**Comment by astrojuanlu** [comment]
> > * Right after the initial task finishes, `kedro_datasets` is not installed.
> 
> Do we really need `kedro_datasets` installed for `kedro` or is it required by building docs?

They're needed to build the docs.

> > * `make build-docs` installs `kedro-datasets[all]~=1.4.0` (pulled from the `docs` optional dependencies group) and immediately fails on `VideoDataSet`
> > * Then `sudo apt-get update && sudo apt-get install -y --no-install-recommends libgl1` as advised in https://stackoverflow.com/a/74501248/554319 fixes the import, and `make build-docs` gets back to _almost_ working, but fails when generating the output because `mmdc` (the Mermaid CLI) is not installed
> 
> `sudo` will require the Dockerfile approach? I try running sudo in terminal it doesn't work.

I just tested it and worked fine for me in the terminal 🤔 Could you share a screenshot?

**Comment by astrojuanlu** [comment]
Building a complete version of the docs without warnings requires `kedro-datasets[all]` indeed. It has always been this way actually, but now it's official:

https://github.com/kedro-org/kedro/blob/fd8162d5bf384ef666c01ef2c529d01fd9fa8354/setup.py#L107

That doesn't mean that the Gitpod image should install all of them (after all, `make build-docs` would take care of it).

**Comment by noklam** [comment]
This is the closest I can get now, it failed midway with some mermaid error. 

> Warning, treated as error:
> mermaid code 'flowchart TD\n  A[Start] --> B{Do you prefer developing your projects in notebooks?}\n  B -->|Yes| C[Use a Databricks workspace to develop a Kedro project]\n  B -->|No| D{Are you a beginner with Kedro?}\n  D -->|Yes| E[Use an IDE, dbx and Databricks Repos to develop a Kedro project]\n  D -->|No| F{Do you have advanced project requirements<br>e.g. CI/CD, scheduling, production-ready, complex pipelines, etc.?}\n  F -->|Yes| G{Is rapid development needed for your project needs?}\n  F -->|No| H[Use an IDE, dbx and Databricks Repos to develop a Kedro project]\n  G -->|Yes| I[Use an IDE, dbx and Databricks Repos to develop a Kedro project]\n  G -->|No| J[Use a Databricks job to deploy a Kedro project]': Mermaid exited with error:
> [stderr]
> b'\x1b[1m\x1b[43m\x1b[30m\n  Puppeteer old Headless deprecation warning:\x1b[0m\x1b[33m\n    In the near feature `headless: true` will default to the new Headless mode\n    for Chrome instead of the old Headless implementation. For more\n    information, please see https://developer.chrome.com/articles/new-headless/.\n    Consider opting in early by passing `headless: "new"` to `puppeteer.launch()`\n    If you encounter any bugs, please report them to https://github.com/puppeteer/puppeteer/issues/new/choose.\x1b[0m\n\n\nError: Failed to launch the browser process!\n/home/gitpod/.cache/puppeteer/chrome/linux-1108766/chrome-linux/chrome: error while loading shared libraries: libatk-1.0.so.0: cannot open shared object file: No such file or directory\n\n\nTROUBLESHOOTING: https://pptr.dev/troubleshooting\n\n    at Interface.onClose (file:///home/gitpod/.nvm/versions/node/v18.16.0/lib/node_modules/@mermaid-js/mermaid-cli/node_modules/@puppeteer/browsers/lib/esm/launch.js:253:24)\n    at Interface.emit (node:events:525:35)\n    at Interface.close (node:internal/readline/interface:533:10)\n    at Socket.onend (node:internal/readline/interface:259:10)\n    at Socket.emit (node:events:525:35)\n    at endReadableNT (node:internal/streams/readable:1359:12)\n    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)\n\n'
> [stdout]
> b''

**Comment by astrojuanlu** [comment]
Guessing you didn't want to close this? :smile

**Comment by noklam** [comment]
@astrojuanlu Ah, accidentally used the keywords `fix` and GitHub think it is smart...

**Comment by noklam** [comment]
@astrojuanlu It is working now when I start a new Gitpod and do `make build-docs`.

File: .gitpod.yml [hunk 1]
```diff
@@ -1,32 +1,23 @@
...
+  file: .gitpod.Dockerfile
 tasks:
   # We want packages installed during the pre-build init steps to go to /workspace
   # rather than ~ so that they are persisted. Gitpod sets PIP_USER=yes to ensure this,
   # but pre-commit requires PIP_USER=no. Hence we set PIP_USER=no and use
   # pip install --user to install to /workspace.
   - name: kedro
-    before: |
-      echo PIP_USER=no >> ~/.bashrc && export PIP_USER=no
+    # before: |
```
Comment (astrojuanlu) [inline .gitpod.yml:9]: Let's remove these lines if they're commented out

**Review by astrojuanlu — APPROVED** [review]
LGTM with a nitpick!

**Review by ankatiyar — APPROVED** [review]
I tried `make build-docs` and `make lint` and it's working!

**Pull request merged**